src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslCertificatePal.cs

```diff
@@ -0,0 +1,203 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed partial class OpenSslCertificatePal
```

Comment:
All of the methods added here should just go on OpenSslCertificateReader.  The macOS and Windows types that end in "CertificatePal" are the ICertificatePal implementations, having a "-CertificatePal" type here that isn't is more confusing than it just having a different name.